OMG, awesome PR. Thanks! :)

The only thing left to fix is make the `ruff` check pass on the pre-commit.ci.

Here's the output:

```bash
Ruff check...............................................................Failed
- hook id: ruff
- exit code: 1

warning: `one-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `one-blank-line-before-class`.
warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
tests/test_general.py:251:16: E712 Avoid equality comparisons to `True`; use `if test():` for truth checks
    |
249 |     cachier.disable_caching()
250 |     try:
251 |         assert test() == True
    |                ^^^^^^^^^^^^^^ E712
252 |     finally:
253 |         cachier.enable_caching()
    |
    = help: Replace with `test()`

tests/test_general.py:264:16: E712 Avoid equality comparisons to `True`; use `if Test().test():` for truth checks
    |
262 |     cachier.disable_caching()
263 |     try:
264 |         assert Test().test() == True
    |                ^^^^^^^^^^^^^^^^^^^^^ E712
265 |     finally:
266 |         cachier.enable_caching()
    |
    = help: Replace with `Test().test()`

Found 2 errors.
No fixes available (2 hidden fixes can be enabled with the `--unsafe-fixes` option).
```

--- a/src/cachier/core.py
+++ b/src/cachier/core.py
@@ -245,7 +245,11 @@ def cachier(
             if verbose:
                 _print = print
             if ignore_cache or not _default_params["caching_enabled"]:
-                return func(**kwargs)
+                return (
+                    func(args[0], **kwargs)
+                    if core.func_is_method
+                    else func(**kwargs)
+                )
             key, entry = core.get_entry((), kwargs)
             if overwrite_cache:
                 return _calc_entry(core, key, func, args, kwds)

--- a/tests/test_general.py
+++ b/tests/test_general.py
@@ -241,6 +241,76 @@ def test_global_disable():
     assert result_1 != result_3
 
 
+def test_global_disable_function():
+    @cachier.cachier()
+    def test():
+        return True
+
+    cachier.disable_caching()
+    try:
+        assert test()
+    finally:
+        cachier.enable_caching()
+
+
+def test_global_disable_method():
+    class Test:
+        @cachier.cachier()
+        def test(self):
+            return True
+
+    cachier.disable_caching()
+    try:
+        assert Test().test()
+    finally:
+        cachier.enable_caching()
+
+
+def test_global_disable_method_with_args():
+    class Test:
+        @cachier.cachier()
+        def test(self, test):
+            return test
+
+    cachier.disable_caching()
+    try:
+        assert Test().test(1) == 1
+    finally:
+        cachier.enable_caching()
+
+
+def test_global_disable_method_with_optional_parameters():
+    class Test:
+        def __init__(self, val):
+            self.val = val
+
+        @cachier.cachier()
+        def test(self, test=0):
+            return self.val + test
+
+    cachier.disable_caching()
+    try:
+        assert Test(1).test(test=1) == 2
+    finally:
+        cachier.enable_caching()
+
+
+def test_global_disable_method_with_args_and_optional_parameters():
+    class Test:
+        def __init__(self, val):
+            self.val = val
+
+        @cachier.cachier()
+        def test(self, test1, test2=0):
+            return self.val + test1 + test2
+
+    cachier.disable_caching()
+    try:
+        assert Test(1).test(2, 3) == 6
+    finally:
+        cachier.enable_caching()
+
+
 def test_none_not_cached_by_default():
     count = 0
 

